tests/integration/cloudformation/resources/test_kinesis.snapshot.json

```diff
@@ -29,11 +29,6 @@
       "stream_description": {
         "StreamDescription": {
           "EncryptionType": "NONE",
-          "EnhancedMonitoring": [
```

Comment:
These are not returned when empty now. According to the Kinesis API, ShardLevelMetrics must be non-empty if this is returned, otherwise it is null. In this case, it is null.

Reply:
Hi @etspaceman !

These are actually validated against AWS. Our `*.snapshot.json` files are all automatically generated when running against AWS and we generally don't update them manually for this reason. To update them we run the test with the following additional environment variables.
```
AWS_PROFILE=<your-profile>
TEST_TARGET=AWS_CLOUD
SNAPSHOT_UPDATE=1  # or set it to `0` if you just want to verify that the existing snapshot still validates against AWS
```

This could've changed since we recorded it, but I just updated to have the newest specs in boto3, re-ran the test and this seems to still be the observed result unfortunately.

Additionally, please note that these are not the raw HTTP responses you're seeing, but the objects generated by the SDK.

Reply:
That's very interesting. Here is section in the smithy documents about this:

https://github.com/aws/aws-sdk-js-v3/blob/main/codegen/sdk-codegen/aws-models/kinesis.json#L5072

You can see that the min length here is 1. I've confirmed this in the Java SDK as well.

But also, if I run a describe-stream in the aws-cli, I see empty lists being returned. So I wonder if this is a bug in Smithy.

Reply:
@dominikschubert I am going to open a support ticket with AWS on this one to get some clarity. For now I'll close this PR and revert my change in kinesis-mock